// @fluentui/make-styles/lib-commonjs/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.__styles = exports.resolveStyleRules = exports.resolveProxyValues = exports.createCSSVariablesProxy = exports.resolveStyleRulesForSlots = exports.makeStyles = exports.makeStaticStyles = exports.mergeClasses = exports.rehydrateRendererCache = exports.styleBucketOrdering = exports.createDOMRenderer = void 0;

var tslib_1 = /*#__PURE__*/require("tslib");

var createDOMRenderer_1 = /*#__PURE__*/require("./renderer/createDOMRenderer");

Object.defineProperty(exports, "createDOMRenderer", {
  enumerable: true,
  get: function () {
    return createDOMRenderer_1.createDOMRenderer;
  }
});

var getStyleSheetForBucket_1 = /*#__PURE__*/require("./renderer/getStyleSheetForBucket");

Object.defineProperty(exports, "styleBucketOrdering", {
  enumerable: true,
  get: function () {
    return getStyleSheetForBucket_1.styleBucketOrdering;
  }
});

var rehydrateRendererCache_1 = /*#__PURE__*/require("./renderer/rehydrateRendererCache");

Object.defineProperty(exports, "rehydrateRendererCache", {
  enumerable: true,
  get: function () {
    return rehydrateRendererCache_1.rehydrateRendererCache;
  }
});

var mergeClasses_1 = /*#__PURE__*/require("./mergeClasses");

Object.defineProperty(exports, "mergeClasses", {
  enumerable: true,
  get: function () {
    return mergeClasses_1.mergeClasses;
  }
});

var makeStaticStyles_1 = /*#__PURE__*/require("./makeStaticStyles");

Object.defineProperty(exports, "makeStaticStyles", {
  enumerable: true,
  get: function () {
    return makeStaticStyles_1.makeStaticStyles;
  }
});

var makeStyles_1 = /*#__PURE__*/require("./makeStyles");

Object.defineProperty(exports, "makeStyles", {
  enumerable: true,
  get: function () {
    return makeStyles_1.makeStyles;
  }
});

var resolveStyleRulesForSlots_1 = /*#__PURE__*/require("./resolveStyleRulesForSlots");

Object.defineProperty(exports, "resolveStyleRulesForSlots", {
  enumerable: true,
  get: function () {
    return resolveStyleRulesForSlots_1.resolveStyleRulesForSlots;
  }
}); // Private exports, are used by build time transforms

var createCSSVariablesProxy_1 = /*#__PURE__*/require("./runtime/createCSSVariablesProxy");

Object.defineProperty(exports, "createCSSVariablesProxy", {
  enumerable: true,
  get: function () {
    return createCSSVariablesProxy_1.createCSSVariablesProxy;
  }
});
Object.defineProperty(exports, "resolveProxyValues", {
  enumerable: true,
  get: function () {
    return createCSSVariablesProxy_1.resolveProxyValues;
  }
});

var resolveStyleRules_1 = /*#__PURE__*/require("./runtime/resolveStyleRules");

Object.defineProperty(exports, "resolveStyleRules", {
  enumerable: true,
  get: function () {
    return resolveStyleRules_1.resolveStyleRules;
  }
});

var __styles_1 = /*#__PURE__*/require("./__styles");

Object.defineProperty(exports, "__styles", {
  enumerable: true,
  get: function () {
    return __styles_1.__styles;
  }
});

tslib_1.__exportStar(require("./types"), exports);

tslib_1.__exportStar(require("./constants"), exports);
