// @fluentui/react-switch/lib-commonjs/components/Switch/useSwitch.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useSwitch = exports.switchShorthandProps = void 0;

var tslib_1 = /*#__PURE__*/require("tslib");

var react_utilities_1 = /*#__PURE__*/require("@fluentui/react-utilities");

var useSwitchState_1 = /*#__PURE__*/require("./useSwitchState");
/**
 * Array of all shorthand properties listed in SwitchSlots
 */


exports.switchShorthandProps = ['root', 'track', 'thumbWrapper', 'thumb', 'activeRail', 'input'];
/**
 * Given user props, returns state and render function for a Switch.
 */

var useSwitch = function (props, ref) {
  var track = props.track,
      thumbWrapper = props.thumbWrapper,
      thumb = props.thumb,
      activeRail = props.activeRail,
      input = props.input,
      defaultChecked = props.defaultChecked,
      checked = props.checked,
      disabled = props.disabled,
      onChange = props.onChange;
  var state = {
    defaultChecked: defaultChecked,
    checked: checked,
    disabled: disabled,
    onChange: onChange,
    root: react_utilities_1.getNativeElementProps('span', tslib_1.__assign(tslib_1.__assign({
      ref: ref
    }, props), {
      id: react_utilities_1.useId('switch-', props.id)
    })),
    components: {
      root: 'div',
      track: 'div',
      thumbWrapper: 'div',
      thumb: 'div',
      activeRail: 'div',
      input: 'input'
    },
    track: react_utilities_1.resolveShorthand(track, {
      required: true
    }),
    thumbWrapper: react_utilities_1.resolveShorthand(thumbWrapper, {
      required: true
    }),
    thumb: react_utilities_1.resolveShorthand(thumb, {
      required: true
    }),
    activeRail: react_utilities_1.resolveShorthand(activeRail, {
      required: true
    }),
    input: react_utilities_1.resolveShorthand(input, {
      required: true,
      defaultProps: {
        type: 'checkbox'
      }
    })
  };
  useSwitchState_1.useSwitchState(state);
  return state;
};

exports.useSwitch = useSwitch;
