// @pongoai/react-textfield/lib-commonjs/components/Password/usePassword.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.usePassword = exports.passwordShorthandProps = void 0;
const react_utilities_1 = require("@fluentui/react-utilities");
const usePasswordState_1 = require("./usePasswordState");
exports.passwordShorthandProps = [
    'root',
    'input',
    'inputWrapper',
    'textFieldLabel',
    'textFieldLegend',
    'textFieldHelperText',
    'showIconWrapper',
    'textFieldWrapper',
    'textFieldBorder',
];
const usePassword = (props, ref) => {
    const { defaultValue = '', value, autocomplete, disabled = false, error = false, appearance = 'standard', placeholder, required = false, label, helperText, input, inputWrapper, textFieldHelperText, showIconWrapper, textFieldLabel, textFieldLegend, textFieldBorder, textFieldWrapper, onChange, } = props;
    const state = {
        defaultValue,
        value,
        autocomplete,
        disabled,
        error,
        appearance,
        placeholder,
        helperText,
        required,
        label,
        onChange,
        root: react_utilities_1.getNativeElementProps('div', {
            ref,
            ...props,
        }),
        components: {
            root: 'div',
            textFieldBorder: 'fieldset',
            textFieldLegend: 'legend',
            textFieldHelperText: 'p',
            textFieldWrapper: 'div',
            textFieldLabel: 'label',
            showIconWrapper: 'span',
            inputWrapper: 'div',
            input: 'input',
        },
        textFieldBorder: react_utilities_1.resolveShorthand(textFieldBorder, { required: true }),
        textFieldLegend: react_utilities_1.resolveShorthand(textFieldLegend, { required: true }),
        textFieldWrapper: react_utilities_1.resolveShorthand(textFieldWrapper, { required: true }),
        showIconWrapper: react_utilities_1.resolveShorthand(showIconWrapper, { required: true }),
        textFieldLabel: react_utilities_1.resolveShorthand(textFieldLabel, { required: true }),
        textFieldHelperText: react_utilities_1.resolveShorthand(textFieldHelperText, { required: true }),
        inputWrapper: react_utilities_1.resolveShorthand(inputWrapper, { required: true }),
        input: react_utilities_1.resolveShorthand(input, {
            required: true,
            defaultProps: {
                type: 'password',
            },
        }),
    };
    usePasswordState_1.usePasswordState(state);
    return state;
};
exports.usePassword = usePassword;
