// @fluentui/react-theme/lib-commonjs/alias/light.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.generateSharedColorTokens = exports.generateNeutralColorTokens = void 0;

var generateNeutralColorTokens = function (global) {
  return {
    neutralForeground1: global.palette.grey[14],
    neutralForeground2: global.palette.grey[26],
    neutralForeground2Hover: global.palette.grey[14],
    neutralForeground2Pressed: global.palette.grey[14],
    neutralForeground2Selected: global.palette.grey[14],
    neutralForeground2BrandHover: global.palette.brand.primary,
    neutralForeground2BrandPressed: global.palette.brand.shade10,
    neutralForeground2BrandSelected: global.palette.brand.primary,
    neutralForeground3: global.palette.grey[38],
    neutralForeground3Hover: global.palette.grey[26],
    neutralForeground3Pressed: global.palette.grey[26],
    neutralForeground3Selected: global.palette.grey[26],
    neutralForeground3BrandHover: global.palette.brand.primary,
    neutralForeground3BrandPressed: global.palette.brand.shade10,
    neutralForeground3BrandSelected: global.palette.brand.primary,
    neutralForeground4: global.palette.grey[44],
    neutralForegroundDisabled: global.palette.grey[74],
    brandForegroundLink: global.palette.brand.shade10,
    brandForegroundLinkHover: global.palette.brand.shade20,
    brandForegroundLinkPressed: global.palette.brand.shade40,
    brandForegroundLinkSelected: global.palette.brand.shade10,
    compoundBrandForeground1: global.palette.brand.primary,
    compoundBrandForeground1Hover: global.palette.brand.shade10,
    compoundBrandForeground1Pressed: global.palette.brand.shade20,
    brandForeground1: global.palette.brand.primary,
    brandForeground2: global.palette.brand.shade10,
    neutralForegroundInverted: global.color.white,
    neutralForegroundOnBrand: global.color.white,
    neutralForegroundInvertedLink: global.color.white,
    neutralForegroundInvertedLinkHover: global.color.white,
    neutralForegroundInvertedLinkPressed: global.color.white,
    neutralForegroundInvertedLinkSelected: global.color.white,
    neutralBackground1: global.color.white,
    neutralBackground1Hover: global.palette.grey[96],
    neutralBackground1Pressed: global.palette.grey[88],
    neutralBackground1Selected: global.palette.grey[92],
    neutralBackground2: global.palette.grey[98],
    neutralBackground2Hover: global.palette.grey[94],
    neutralBackground2Pressed: global.palette.grey[86],
    neutralBackground2Selected: global.palette.grey[90],
    neutralBackground3: global.palette.grey[96],
    neutralBackground3Hover: global.palette.grey[92],
    neutralBackground3Pressed: global.palette.grey[84],
    neutralBackground3Selected: global.palette.grey[88],
    neutralBackground4: global.palette.grey[94],
    neutralBackground4Hover: global.palette.grey[98],
    neutralBackground4Pressed: global.palette.grey[96],
    neutralBackground4Selected: global.color.white,
    neutralBackground5: global.palette.grey[92],
    neutralBackground5Hover: global.palette.grey[96],
    neutralBackground5Pressed: global.palette.grey[94],
    neutralBackground5Selected: global.palette.grey[98],
    neutralBackground6: global.palette.grey[90],
    neutralBackgroundInverted: global.palette.grey[38],
    subtleBackground: 'transparent',
    subtleBackgroundHover: global.palette.grey[96],
    subtleBackgroundPressed: global.palette.grey[88],
    subtleBackgroundSelected: global.palette.grey[92],
    transparentBackground: 'transparent',
    transparentBackgroundHover: 'transparent',
    transparentBackgroundPressed: 'transparent',
    transparentBackgroundSelected: 'transparent',
    neutralBackgroundDisabled: global.palette.grey[94],
    neutralStencil1: global.palette.grey[90],
    neutralStencil2: global.palette.grey[98],
    brandBackground: global.palette.brand.primary,
    brandBackgroundHover: global.palette.brand.shade10,
    brandBackgroundPressed: global.palette.brand.shade40,
    brandBackgroundSelected: global.palette.brand.shade20,
    compoundBrandBackground: global.palette.brand.primary,
    compoundBrandBackgroundHover: global.palette.brand.shade10,
    compoundBrandBackgroundPressed: global.palette.brand.shade20,
    brandBackgroundStatic: global.palette.brand.primary,
    brandBackground2: global.palette.brand.tint60,
    neutralStrokeAccessible: global.palette.grey[38],
    neutralStrokeAccessibleHover: global.palette.grey[34],
    neutralStrokeAccessiblePressed: global.palette.grey[30],
    neutralStrokeAccessibleSelected: global.palette.brand.primary,
    neutralStroke1: global.palette.grey[82],
    neutralStroke1Hover: global.palette.grey[78],
    neutralStroke1Pressed: global.palette.grey[70],
    neutralStroke1Selected: global.palette.grey[74],
    neutralStroke2: global.palette.grey[88],
    neutralStroke3: global.palette.grey[94],
    brandStroke1: global.palette.brand.primary,
    brandStroke2: global.palette.brand.tint40,
    compoundBrandStroke: global.palette.brand.primary,
    compoundBrandStrokeHover: global.palette.brand.shade10,
    compoundBrandStrokePressed: global.palette.brand.shade20,
    neutralStrokeDisabled: global.palette.grey[88],
    transparentStroke: 'transparent',
    transparentStrokeInteractive: 'transparent',
    transparentStrokeDisabled: 'transparent',
    strokeFocus1: global.color.white,
    strokeFocus2: global.color.black,
    neutralShadowAmbient: 'rgba(0,0,0,0.12)',
    neutralShadowKey: 'rgba(0,0,0,0.14)',
    neutralShadowAmbientLighter: 'rgba(0,0,0,0.06)',
    neutralShadowKeyLighter: 'rgba(0,0,0,0.07)',
    neutralShadowAmbientDarker: 'rgba(0,0,0,0.20)',
    neutralShadowKeyDarker: 'rgba(0,0,0,0.24)',
    brandShadowAmbient: 'rgba(0,0,0,0.30)',
    brandShadowKey: 'rgba(0,0,0,0.25)'
  };
};

exports.generateNeutralColorTokens = generateNeutralColorTokens;

var generateSharedColorTokens = function (globalSharedTokens) {
  return Object.keys(globalSharedTokens).reduce(function (acc, sharedColor) {
    acc[sharedColor] = {
      background1: globalSharedTokens[sharedColor].tint60,
      background2: globalSharedTokens[sharedColor].tint40,
      background3: globalSharedTokens[sharedColor].primary,
      foreground1: globalSharedTokens[sharedColor].shade10,
      foreground2: globalSharedTokens[sharedColor].shade30,
      foreground3: globalSharedTokens[sharedColor].primary,
      borderActive: globalSharedTokens[sharedColor].primary,
      border2: globalSharedTokens[sharedColor].primary
    };
    return acc;
  }, {});
};

exports.generateSharedColorTokens = generateSharedColorTokens;
