// @pongoai/react-rating/lib/Rating.stories.js
import * as React from 'react';
import { Rating } from './index';
import { Header3 } from '@pongoai/react-text';
import { makeStyles } from '@fluentui/react-make-styles';
const useStyles = makeStyles({
    root: {
        display: 'inline-flex',
        flexDirection: 'column',
        gap: '10px',
    },
    row: {
        display: 'inline-flex',
        flexDirection: 'column',
        gap: '10px',
    },
});
export const BasicRatingExample = () => {
    const [ratingValue, setRatingValue] = React.useState(3);
    const styles = useStyles();
    const onRatingChange = (_ev, data) => {
        setRatingValue(data.value);
    };
    return (React.createElement("div", { className: styles.root },
        React.createElement(Header3, null, "Basic Example"),
        React.createElement("div", { className: styles.row },
            React.createElement(Rating, null)),
        React.createElement(Header3, null,
            "Controlled Example ",
            ratingValue),
        React.createElement("div", { className: styles.row },
            React.createElement(Rating, { value: ratingValue, onChange: onRatingChange })),
        React.createElement(Header3, null, "Read Only"),
        React.createElement("div", { className: styles.row },
            React.createElement(Rating, { defaultValue: 2, readOnly: true })),
        React.createElement(Header3, null, "Size"),
        React.createElement("div", { className: styles.row },
            React.createElement(Rating, { size: "small" }),
            React.createElement(Rating, { size: "medium" }),
            React.createElement(Rating, { size: "large" })),
        React.createElement(Header3, null, "Multiple Stars"),
        React.createElement("div", { className: styles.row },
            React.createElement(Rating, { defaultValue: 10, max: 10 })),
        React.createElement(Header3, null, "Disabled"),
        React.createElement("div", { className: styles.row },
            React.createElement(Rating, { defaultValue: 3, disabled: true }))));
};
export default {
    title: 'Components/Rating',
    component: Rating,
};
