// @pongoai/react-textfield/lib-commonjs/Multiline.stories.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.BasicMultilineExample = void 0;
const React = __importStar(require("react"));
const index_1 = require("./index");
const react_text_1 = require("@pongoai/react-text");
const react_make_styles_1 = require("@fluentui/react-make-styles");
const useStyles = react_make_styles_1.makeStyles({
    root: {
        display: 'inline-flex',
        flexDirection: 'column',
        gap: '5px',
    },
    row: {
        display: 'inline-flex',
        flexDirection: 'row',
        gap: '10px',
    },
});
const BasicMultilineExample = () => {
    const styles = useStyles();
    return (React.createElement("div", { className: styles.root },
        React.createElement(react_text_1.Header2, null, "Standard Multiline Text Field"),
        React.createElement("div", { className: styles.row },
            React.createElement(index_1.Multiline, { label: "Standard", placeholder: "Enter your text" }),
            React.createElement(index_1.Multiline, { label: "Standard", placeholder: "Enter your text", disabled: true }),
            React.createElement(index_1.Multiline, { label: "Standard", placeholder: "Enter your text", helperText: "Error", error: true })),
        React.createElement(react_text_1.Header2, null, "Outlined Multiline Text Field"),
        React.createElement("div", { className: styles.row },
            React.createElement(index_1.Multiline, { appearance: "outlined", placeholder: "Enter your text", label: "Outlined" }),
            React.createElement(index_1.TextField, { appearance: "outlined", label: "Outlined", placeholder: "Enter your text" }),
            React.createElement(index_1.Multiline, { appearance: "outlined", placeholder: "Enter your text", label: "Outlined", disabled: true }),
            React.createElement(index_1.Multiline, { appearance: "outlined", label: "Standard", placeholder: "Enter your text", helperText: "Error", error: true })),
        React.createElement(react_text_1.Header2, null, "Filled Multiline Text Field"),
        React.createElement("div", { className: styles.row },
            React.createElement(index_1.Multiline, { appearance: "filled", placeholder: "Enter your text", label: "Filled" }),
            React.createElement(index_1.Multiline, { appearance: "filled", placeholder: "Enter your text", label: "Filled", disabled: true }),
            React.createElement(index_1.Multiline, { appearance: "filled", label: "Standard", placeholder: "Enter your text", helperText: "Error", error: true }))));
};
exports.BasicMultilineExample = BasicMultilineExample;
exports.default = {
    title: 'Components/Multiline',
    component: index_1.Multiline,
};
