// @pongoai/react-textfield/lib-commonjs/components/Search/useSearch.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.useSearch = exports.searchShorthandProps = void 0;
const react_utilities_1 = require("@fluentui/react-utilities");
const useSearchState_1 = require("./useSearchState");
exports.searchShorthandProps = [
    'root',
    'input',
    'inputWrapper',
    'textFieldLabel',
    'textFieldLegend',
    'textFieldHelperText',
    'searchIconWrapper',
    'clearIconWrapper',
    'textFieldWrapper',
    'textFieldBorder',
];
const useSearch = (props, ref) => {
    const { defaultValue = '', value, autocomplete, disabled = false, error = false, appearance = 'standard', placeholder, required = false, label, helperText, input, inputWrapper, textFieldHelperText, searchIconWrapper, clearIconWrapper, textFieldLabel, textFieldLegend, textFieldBorder, textFieldWrapper, onChange, } = props;
    const state = {
        defaultValue,
        value,
        autocomplete,
        disabled,
        error,
        appearance,
        placeholder,
        helperText,
        required,
        label,
        onChange,
        root: react_utilities_1.getNativeElementProps('div', {
            ref,
            ...props,
        }),
        components: {
            root: 'div',
            textFieldBorder: 'fieldset',
            textFieldLegend: 'legend',
            textFieldHelperText: 'p',
            textFieldWrapper: 'div',
            textFieldLabel: 'label',
            searchIconWrapper: 'span',
            clearIconWrapper: 'span',
            inputWrapper: 'div',
            input: 'input',
        },
        textFieldBorder: react_utilities_1.resolveShorthand(textFieldBorder, { required: true }),
        textFieldLegend: react_utilities_1.resolveShorthand(textFieldLegend, { required: true }),
        textFieldWrapper: react_utilities_1.resolveShorthand(textFieldWrapper, { required: true }),
        searchIconWrapper: react_utilities_1.resolveShorthand(searchIconWrapper, { required: true }),
        clearIconWrapper: react_utilities_1.resolveShorthand(clearIconWrapper, { required: true }),
        textFieldLabel: react_utilities_1.resolveShorthand(textFieldLabel, { required: true }),
        textFieldHelperText: react_utilities_1.resolveShorthand(textFieldHelperText, { required: true }),
        inputWrapper: react_utilities_1.resolveShorthand(inputWrapper, { required: true }),
        input: react_utilities_1.resolveShorthand(input, {
            required: true,
            defaultProps: {
                type: 'search',
            },
        }),
    };
    useSearchState_1.useSearchState(state);
    return state;
};
exports.useSearch = useSearch;
