// @pongoai/react-text/lib-commonjs/Text.stories.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.BasicTextExample = void 0;
const React = __importStar(require("react"));
const index_1 = require("./index");
const react_make_styles_1 = require("@fluentui/react-make-styles");
const useStyles = react_make_styles_1.makeStyles({
    root: {
        display: 'inline-flex',
        flexDirection: 'column',
        // gap: '10px',
    },
    row: {
        display: 'inline-flex',
        flexDirection: 'row',
        // gap: '10px',
    },
});
const BasicTextExample = () => {
    const styles = useStyles();
    return (React.createElement("div", { className: styles.root },
        React.createElement(index_1.Text, { size: 1000 }, "Header"),
        React.createElement("div", { className: styles.row },
            React.createElement(index_1.Text, null, "Basic"),
            React.createElement(index_1.Text, { italic: true }, "Italic"),
            React.createElement(index_1.Text, { weight: "bold" }, "Weight"),
            React.createElement(index_1.Text, { font: "monospace" }, "monospace")),
        React.createElement("div", { className: styles.row },
            React.createElement(index_1.Text, { strikethrough: true }, "Strikethrough"),
            React.createElement(index_1.Text, { underline: true, strikethrough: true }, "Underline"),
            React.createElement(index_1.Text, { underline: true, strikethrough: true }, "Underline Strikethrough")),
        React.createElement(index_1.Display, null, "Display: The quick brown fox jumps over the lazy dog"),
        React.createElement(index_1.Title, null, "Title: The quick brown fox jumps over the lazy dog"),
        React.createElement(index_1.Header1, null, "Header1: The quick brown fox jumps over the lazy dog"),
        React.createElement(index_1.Header2, null, "Header2: The quick brown fox jumps over the lazy dog"),
        React.createElement(index_1.Header3, null, "Header3: The quick brown fox jumps over the lazy dog"),
        React.createElement(index_1.SubHeadline, null, "SubHeadline: The quick brown fox jumps over the lazy dog"),
        React.createElement(index_1.Body, null, "Body: The quick brown fox jumps over the lazy dog"),
        React.createElement(index_1.Caption, null, "Caption: The quick brown fox jumps over the lazy dog")));
};
exports.BasicTextExample = BasicTextExample;
exports.default = {
    title: 'Components/Text',
    component: index_1.Text,
};
