// @pongoai/react-textfield/lib/components/Multiline/useMultiline.js
import { getNativeElementProps, resolveShorthand } from '@fluentui/react-utilities';
import { useMultilineState } from './useMultilineState';
export const multilineShorthandProps = [
    'root',
    'textarea',
    'textFieldLabel',
    'textFieldLegend',
    'textFieldHelperText',
    'textFieldBorder',
];
export const useMultiline = (props, ref) => {
    const { defaultValue = '', value, autocomplete, disabled = false, error = false, appearance = 'standard', textarea, placeholder, required = false, autoAdjust = true, label, helperText, textFieldHelperText, textFieldLabel, textFieldLegend, textFieldBorder, resizable = false, onChange, } = props;
    const state = {
        defaultValue,
        value,
        autocomplete,
        disabled,
        error,
        appearance,
        placeholder,
        resizable,
        helperText,
        autoAdjust,
        required,
        label,
        onChange,
        root: getNativeElementProps('div', {
            ref,
            ...props,
        }),
        components: {
            root: 'div',
            textFieldBorder: 'fieldset',
            textFieldLegend: 'legend',
            textFieldHelperText: 'p',
            textFieldLabel: 'label',
            textarea: 'textarea',
        },
        textFieldBorder: resolveShorthand(textFieldBorder, { required: true }),
        textFieldLegend: resolveShorthand(textFieldLegend, { required: true }),
        textFieldLabel: resolveShorthand(textFieldLabel, { required: true }),
        textFieldHelperText: resolveShorthand(textFieldHelperText, { required: true }),
        textarea: resolveShorthand(textarea, {
            required: true,
        }),
    };
    useMultilineState(state);
    return state;
};
