// @fluentui/react-theme/lib/alias/dark.js
export var generateNeutralColorTokens = function (global) {
  return {
    neutralForeground1: global.color.white,
    neutralForeground2: global.palette.grey[84],
    neutralForeground2Hover: global.color.white,
    neutralForeground2Pressed: global.color.white,
    neutralForeground2Selected: global.color.white,
    neutralForeground2BrandHover: global.palette.brand.tint20,
    neutralForeground2BrandPressed: global.palette.brand.tint10,
    neutralForeground2BrandSelected: global.palette.brand.tint20,
    neutralForeground3: global.palette.grey[68],
    neutralForeground3Hover: global.palette.grey[84],
    neutralForeground3Pressed: global.palette.grey[84],
    neutralForeground3Selected: global.palette.grey[84],
    neutralForeground3BrandHover: global.palette.brand.tint20,
    neutralForeground3BrandPressed: global.palette.brand.tint10,
    neutralForeground3BrandSelected: global.palette.brand.tint20,
    neutralForeground4: global.palette.grey[60],
    neutralForegroundDisabled: global.palette.grey[36],
    brandForegroundLink: global.palette.brand.tint10,
    brandForegroundLinkHover: global.palette.brand.tint30,
    brandForegroundLinkPressed: global.palette.brand.tint20,
    brandForegroundLinkSelected: global.palette.brand.tint10,
    compoundBrandForeground1: global.palette.brand.tint10,
    compoundBrandForeground1Hover: global.palette.brand.tint20,
    compoundBrandForeground1Pressed: global.palette.brand.primary,
    brandForeground1: global.palette.brand.tint20,
    brandForeground2: global.palette.brand.tint30,
    neutralForegroundInverted: global.palette.grey[14],
    neutralForegroundOnBrand: global.color.white,
    neutralForegroundInvertedLink: global.color.white,
    neutralForegroundInvertedLinkHover: global.color.white,
    neutralForegroundInvertedLinkPressed: global.color.white,
    neutralForegroundInvertedLinkSelected: global.color.white,
    neutralBackground1: global.palette.grey[16],
    neutralBackground1Hover: global.palette.grey[24],
    neutralBackground1Pressed: global.palette.grey[12],
    neutralBackground1Selected: global.palette.grey[22],
    neutralBackground2: global.palette.grey[12],
    neutralBackground2Hover: global.palette.grey[20],
    neutralBackground2Pressed: global.palette.grey[8],
    neutralBackground2Selected: global.palette.grey[18],
    neutralBackground3: global.palette.grey[8],
    neutralBackground3Hover: global.palette.grey[16],
    neutralBackground3Pressed: global.palette.grey[4],
    neutralBackground3Selected: global.palette.grey[14],
    neutralBackground4: global.palette.grey[4],
    neutralBackground4Hover: global.palette.grey[12],
    neutralBackground4Pressed: global.color.black,
    neutralBackground4Selected: global.palette.grey[10],
    neutralBackground5: global.color.black,
    neutralBackground5Hover: global.palette.grey[8],
    neutralBackground5Pressed: global.palette.grey[2],
    neutralBackground5Selected: global.palette.grey[6],
    neutralBackground6: global.palette.grey[20],
    neutralBackgroundInverted: global.color.white,
    subtleBackground: 'transparent',
    subtleBackgroundHover: global.palette.grey[22],
    subtleBackgroundPressed: global.palette.grey[18],
    subtleBackgroundSelected: global.palette.grey[20],
    transparentBackground: 'transparent',
    transparentBackgroundHover: 'transparent',
    transparentBackgroundPressed: 'transparent',
    transparentBackgroundSelected: 'transparent',
    neutralBackgroundDisabled: global.palette.grey[8],
    neutralStencil1: global.palette.grey[20],
    neutralStencil2: global.palette.grey[34],
    brandBackground: global.palette.brand.shade10,
    brandBackgroundHover: global.palette.brand.primary,
    brandBackgroundPressed: global.palette.brand.shade40,
    brandBackgroundSelected: global.palette.brand.shade20,
    compoundBrandBackground: global.palette.brand.tint10,
    compoundBrandBackgroundHover: global.palette.brand.tint20,
    compoundBrandBackgroundPressed: global.palette.brand.primary,
    brandBackgroundStatic: global.palette.brand.primary,
    brandBackground2: global.palette.brand.shade40,
    neutralStrokeAccessible: global.palette.grey[68],
    neutralStrokeAccessibleHover: global.palette.grey[74],
    neutralStrokeAccessiblePressed: global.palette.grey[70],
    neutralStrokeAccessibleSelected: global.palette.brand.tint20,
    neutralStroke1: global.palette.grey[40],
    neutralStroke1Hover: global.palette.grey[46],
    neutralStroke1Pressed: global.palette.grey[42],
    neutralStroke1Selected: global.palette.grey[44],
    neutralStroke2: global.palette.grey[32],
    neutralStroke3: global.palette.grey[24],
    brandStroke1: global.palette.brand.tint10,
    brandStroke2: global.palette.brand.shade30,
    compoundBrandStroke: global.palette.brand.tint10,
    compoundBrandStrokeHover: global.palette.brand.tint20,
    compoundBrandStrokePressed: global.palette.brand.primary,
    neutralStrokeDisabled: global.palette.grey[26],
    transparentStroke: 'transparent',
    transparentStrokeInteractive: 'transparent',
    transparentStrokeDisabled: 'transparent',
    strokeFocus1: global.color.black,
    strokeFocus2: global.color.white,
    neutralShadowAmbient: 'rgba(0,0,0,0.24)',
    neutralShadowKey: 'rgba(0,0,0,0.28)',
    neutralShadowAmbientLighter: 'rgba(0,0,0,0.12)',
    neutralShadowKeyLighter: 'rgba(0,0,0,0.14)',
    neutralShadowAmbientDarker: 'rgba(0,0,0,0.40)',
    neutralShadowKeyDarker: 'rgba(0,0,0,0.48)',
    brandShadowAmbient: 'rgba(0,0,0,0.30)',
    brandShadowKey: 'rgba(0,0,0,0.25)'
  };
};
export var generateSharedColorTokens = function (globalSharedTokens) {
  return Object.keys(globalSharedTokens).reduce(function (acc, sharedColor) {
    acc[sharedColor] = {
      background1: globalSharedTokens[sharedColor].shade40,
      background2: globalSharedTokens[sharedColor].shade30,
      background3: globalSharedTokens[sharedColor].primary,
      foreground1: globalSharedTokens[sharedColor].tint30,
      foreground2: globalSharedTokens[sharedColor].tint40,
      foreground3: globalSharedTokens[sharedColor].tint20,
      borderActive: globalSharedTokens[sharedColor].tint30,
      border2: globalSharedTokens[sharedColor].tint20
    };
    return acc;
  }, {});
};
