// @fluentui/react-theme/lib/alias/highContrast.js
export var generateNeutralColorTokens = function (global) {
  return {
    neutralForeground1: global.color.hcCanvasText,
    neutralForeground2: global.color.hcCanvasText,
    neutralForeground2Hover: global.color.hcHighlightText,
    neutralForeground2Pressed: global.color.hcHighlightText,
    neutralForeground2Selected: global.color.hcHighlightText,
    neutralForeground2BrandHover: global.color.hcHighlightText,
    neutralForeground2BrandPressed: global.color.hcHighlightText,
    neutralForeground2BrandSelected: global.color.hcHighlightText,
    neutralForeground3: global.color.hcCanvasText,
    neutralForeground3Hover: global.color.hcHighlightText,
    neutralForeground3Pressed: global.color.hcHighlightText,
    neutralForeground3Selected: global.color.hcHighlightText,
    neutralForeground3BrandHover: global.color.hcHighlightText,
    neutralForeground3BrandPressed: global.color.hcHighlightText,
    neutralForeground3BrandSelected: global.color.hcHighlightText,
    neutralForeground4: global.color.hcCanvasText,
    neutralForegroundDisabled: global.color.hcDisabled,
    brandForegroundLink: global.color.hcHyperlink,
    brandForegroundLinkHover: global.color.hcHyperlink,
    brandForegroundLinkPressed: global.color.hcHyperlink,
    brandForegroundLinkSelected: global.color.hcHyperlink,
    compoundBrandForeground1: global.color.hcHighlight,
    compoundBrandForeground1Hover: global.color.hcHighlight,
    compoundBrandForeground1Pressed: global.color.hcHighlight,
    brandForeground1: global.color.hcCanvasText,
    brandForeground2: global.color.hcButtonText,
    neutralForegroundInverted: global.color.hcCanvas,
    neutralForegroundOnBrand: global.color.hcButtonText,
    neutralForegroundInvertedLink: global.color.hcHyperlink,
    neutralForegroundInvertedLinkHover: global.color.hcHyperlink,
    neutralForegroundInvertedLinkPressed: global.color.hcHyperlink,
    neutralForegroundInvertedLinkSelected: global.color.hcHyperlink,
    neutralBackground1: global.color.hcCanvas,
    neutralBackground1Hover: global.color.hcHighlight,
    neutralBackground1Pressed: global.color.hcHighlight,
    neutralBackground1Selected: global.color.hcHighlight,
    neutralBackground2: global.color.hcCanvas,
    neutralBackground2Hover: global.color.hcHighlight,
    neutralBackground2Pressed: global.color.hcHighlight,
    neutralBackground2Selected: global.color.hcHighlight,
    neutralBackground3: global.color.hcCanvas,
    neutralBackground3Hover: global.color.hcHighlight,
    neutralBackground3Pressed: global.color.hcHighlight,
    neutralBackground3Selected: global.color.hcHighlight,
    neutralBackground4: global.color.hcCanvas,
    neutralBackground4Hover: global.color.hcHighlight,
    neutralBackground4Pressed: global.color.hcHighlight,
    neutralBackground4Selected: global.color.hcHighlight,
    neutralBackground5: global.color.hcCanvas,
    neutralBackground5Hover: global.color.hcHighlight,
    neutralBackground5Pressed: global.color.hcHighlight,
    neutralBackground5Selected: global.color.hcHighlight,
    neutralBackground6: global.color.hcCanvas,
    neutralBackgroundInverted: global.color.hcCanvasText,
    subtleBackground: 'transparent',
    subtleBackgroundHover: global.color.hcHighlight,
    subtleBackgroundPressed: global.color.hcHighlight,
    subtleBackgroundSelected: global.color.hcHighlight,
    transparentBackground: 'transparent',
    transparentBackgroundHover: global.color.hcHighlight,
    transparentBackgroundPressed: global.color.hcHighlight,
    transparentBackgroundSelected: global.color.hcHighlight,
    neutralBackgroundDisabled: global.color.hcCanvas,
    neutralStencil1: global.palette.grey[8],
    neutralStencil2: global.palette.grey[52],
    brandBackground: global.color.hcButtonFace,
    brandBackgroundHover: global.color.hcHighlight,
    brandBackgroundPressed: global.color.hcHighlight,
    brandBackgroundSelected: global.color.hcHighlight,
    compoundBrandBackground: global.color.hcHighlight,
    compoundBrandBackgroundHover: global.color.hcHighlight,
    compoundBrandBackgroundPressed: global.color.hcHighlight,
    brandBackgroundStatic: global.color.hcCanvas,
    brandBackground2: global.color.hcButtonFace,
    neutralStrokeAccessible: global.color.hcCanvasText,
    neutralStrokeAccessibleHover: global.color.hcHighlight,
    neutralStrokeAccessiblePressed: global.color.hcHighlight,
    neutralStrokeAccessibleSelected: global.color.hcHighlight,
    neutralStroke1: global.color.hcCanvasText,
    neutralStroke1Hover: global.color.hcHighlight,
    neutralStroke1Pressed: global.color.hcHighlight,
    neutralStroke1Selected: global.color.hcHighlight,
    neutralStroke2: global.color.hcCanvasText,
    neutralStroke3: global.color.hcCanvasText,
    brandStroke1: global.color.hcCanvasText,
    brandStroke2: global.color.hcCanvas,
    compoundBrandStroke: global.color.hcHighlight,
    compoundBrandStrokeHover: global.color.hcHighlight,
    compoundBrandStrokePressed: global.color.hcHighlight,
    neutralStrokeDisabled: global.color.hcDisabled,
    transparentStroke: global.color.hcCanvasText,
    transparentStrokeInteractive: global.color.hcHighlight,
    transparentStrokeDisabled: global.color.hcDisabled,
    strokeFocus1: global.color.hcCanvas,
    strokeFocus2: global.color.hcHighlight,
    neutralShadowAmbient: 'rgba(0,0,0,0.24)',
    neutralShadowKey: 'rgba(0,0,0,0.28)',
    neutralShadowAmbientLighter: 'rgba(0,0,0,0.12)',
    neutralShadowKeyLighter: 'rgba(0,0,0,0.14)',
    neutralShadowAmbientDarker: 'rgba(0,0,0,0.40)',
    neutralShadowKeyDarker: 'rgba(0,0,0,0.48)',
    brandShadowAmbient: 'rgba(0,0,0,0.30)',
    brandShadowKey: 'rgba(0,0,0,0.25)'
  };
};
export var generateSharedColorTokens = function (globalSharedTokens, globalColors) {
  return Object.keys(globalSharedTokens).reduce(function (acc, sharedColor) {
    acc[sharedColor] = {
      background1: globalColors.white,
      background2: globalColors.black,
      background3: globalColors.white,
      foreground1: globalColors.black,
      foreground2: globalColors.white,
      foreground3: globalColors.white,
      borderActive: globalColors.hcHighlight,
      border2: globalColors.white
    };
    return acc;
  }, {});
};
