// @fluentui/react-theme/lib/global/utils.js
import { __assign } from "tslib";
import { borderRadius } from './borderRadius';
import { black, white, grey, whiteAlpha, blackAlpha, hcButtonFace, hcButtonText, hcCanvas, hcCanvasText, hcDisabled, hcHighlight, hcHighlightText, hcHyperlink } from './colors';
import { textAlignments, fontFamilies, fontWeights, fontSizes, lineHeights } from './fonts';
import { strokeWidths } from './strokeWidths';
export var createGlobalTheme = function (brand, globalSharedColors) {
  return {
    color: {
      white: white,
      black: black,
      hcHyperlink: hcHyperlink,
      hcHighlight: hcHighlight,
      hcDisabled: hcDisabled,
      hcCanvas: hcCanvas,
      hcCanvasText: hcCanvasText,
      hcHighlightText: hcHighlightText,
      hcButtonText: hcButtonText,
      hcButtonFace: hcButtonFace
    },
    palette: __assign(__assign({}, globalSharedColors), {
      brand: brand,
      grey: grey,
      whiteAlpha: whiteAlpha,
      blackAlpha: blackAlpha
    }),
    type: {
      alignment: textAlignments,
      fontFamilies: fontFamilies,
      fontSizes: fontSizes,
      fontWeights: fontWeights,
      lineHeights: lineHeights
    },
    borderRadius: borderRadius,
    strokeWidth: strokeWidths
  };
};
