// @fluentui/react-tabster/lib-commonjs/hooks/useModalAttributes.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useModalAttributes = void 0;

var react_utilities_1 = /*#__PURE__*/require("@fluentui/react-utilities");

var useTabsterAttributes_1 = /*#__PURE__*/require("./useTabsterAttributes");

var tabster_1 = /*#__PURE__*/require("tabster");

var useTabster_1 = /*#__PURE__*/require("./useTabster");
/**
 * Applies modal dialog behaviour through DOM attributes
 * Modal element will focus trap and hide other content on the page
 * The trigger element will be focused if focus is lost after the modal element is removed
 *
 * @returns DOM attributes to apply to the modal element and its trigger
 */


var useModalAttributes = function (options) {
  if (options === void 0) {
    options = {};
  }

  var trapFocus = options.trapFocus,
      alwaysFocusable = options.alwaysFocusable;
  var tabster = useTabster_1.useTabster(); // Initializes the modalizer and deloser APIs

  if (tabster) {
    tabster_1.getModalizer(tabster);
    tabster_1.getDeloser(tabster);
  }

  var id = react_utilities_1.useId('modal-');
  var modalAttributes = useTabsterAttributes_1.useTabsterAttributes({
    deloser: {},
    modalizer: {
      id: id,
      isOthersAccessible: !trapFocus,
      isAlwaysAccessible: alwaysFocusable
    }
  });
  var triggerAttributes = useTabsterAttributes_1.useTabsterAttributes({
    deloser: {}
  });
  return {
    modalAttributes: modalAttributes,
    triggerAttributes: triggerAttributes
  };
};

exports.useModalAttributes = useModalAttributes;
