// @fluentui/react-switch/lib/components/Switch/useSwitch.js
import { __assign } from "tslib";
import { getNativeElementProps, resolveShorthand, useId } from '@fluentui/react-utilities';
import { useSwitchState } from './useSwitchState';
/**
 * Array of all shorthand properties listed in SwitchSlots
 */

export var switchShorthandProps = ['root', 'track', 'thumbWrapper', 'thumb', 'activeRail', 'input'];
/**
 * Given user props, returns state and render function for a Switch.
 */

export var useSwitch = function (props, ref) {
  var track = props.track,
      thumbWrapper = props.thumbWrapper,
      thumb = props.thumb,
      activeRail = props.activeRail,
      input = props.input,
      defaultChecked = props.defaultChecked,
      checked = props.checked,
      disabled = props.disabled,
      onChange = props.onChange;
  var state = {
    defaultChecked: defaultChecked,
    checked: checked,
    disabled: disabled,
    onChange: onChange,
    root: getNativeElementProps('span', __assign(__assign({
      ref: ref
    }, props), {
      id: useId('switch-', props.id)
    })),
    components: {
      root: 'div',
      track: 'div',
      thumbWrapper: 'div',
      thumb: 'div',
      activeRail: 'div',
      input: 'input'
    },
    track: resolveShorthand(track, {
      required: true
    }),
    thumbWrapper: resolveShorthand(thumbWrapper, {
      required: true
    }),
    thumb: resolveShorthand(thumb, {
      required: true
    }),
    activeRail: resolveShorthand(activeRail, {
      required: true
    }),
    input: resolveShorthand(input, {
      required: true,
      defaultProps: {
        type: 'checkbox'
      }
    })
  };
  useSwitchState(state);
  return state;
};
