// @pongoai/react-text/lib/Text.stories.js
import * as React from 'react';
import { Text, Body, Caption, Display, Header1, Header2, Header3, SubHeadline, Title } from './index';
import { makeStyles } from '@fluentui/react-make-styles';
const useStyles = makeStyles({
    root: {
        display: 'inline-flex',
        flexDirection: 'column',
        // gap: '10px',
    },
    row: {
        display: 'inline-flex',
        flexDirection: 'row',
        // gap: '10px',
    },
});
export const BasicTextExample = () => {
    const styles = useStyles();
    return (React.createElement("div", { className: styles.root },
        React.createElement(Text, { size: 1000 }, "Header"),
        React.createElement("div", { className: styles.row },
            React.createElement(Text, null, "Basic"),
            React.createElement(Text, { italic: true }, "Italic"),
            React.createElement(Text, { weight: "bold" }, "Weight"),
            React.createElement(Text, { font: "monospace" }, "monospace")),
        React.createElement("div", { className: styles.row },
            React.createElement(Text, { strikethrough: true }, "Strikethrough"),
            React.createElement(Text, { underline: true, strikethrough: true }, "Underline"),
            React.createElement(Text, { underline: true, strikethrough: true }, "Underline Strikethrough")),
        React.createElement(Display, null, "Display: The quick brown fox jumps over the lazy dog"),
        React.createElement(Title, null, "Title: The quick brown fox jumps over the lazy dog"),
        React.createElement(Header1, null, "Header1: The quick brown fox jumps over the lazy dog"),
        React.createElement(Header2, null, "Header2: The quick brown fox jumps over the lazy dog"),
        React.createElement(Header3, null, "Header3: The quick brown fox jumps over the lazy dog"),
        React.createElement(SubHeadline, null, "SubHeadline: The quick brown fox jumps over the lazy dog"),
        React.createElement(Body, null, "Body: The quick brown fox jumps over the lazy dog"),
        React.createElement(Caption, null, "Caption: The quick brown fox jumps over the lazy dog")));
};
export default {
    title: 'Components/Text',
    component: Text,
};
