// @pongoai/react-textfield/lib-commonjs/components/TextField/useTextField.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.useTextField = exports.textFieldShorthandProps = void 0;
const react_utilities_1 = require("@fluentui/react-utilities");
const useTextFieldState_1 = require("./useTextFieldState");
exports.textFieldShorthandProps = [
    'root',
    'input',
    'inputWrapper',
    'textFieldLabel',
    'textFieldLegend',
    'textFieldHelperText',
    'textFieldSuffix',
    'textFieldPrefix',
    'textFieldWrapper',
    'textFieldBorder',
];
const useTextField = (props, ref) => {
    const { defaultValue = '', value, autocomplete, disabled = false, error = false, appearance = 'standard', input, placeholder, required = false, label, number = false, helperText, suffix, prefix, inputWrapper, textFieldHelperText, textFieldSuffix, textFieldPrefix, textFieldLabel, textFieldLegend, textFieldBorder, textFieldWrapper, onChange, } = props;
    const state = {
        defaultValue,
        value,
        autocomplete,
        disabled,
        error,
        appearance,
        placeholder,
        number,
        suffix,
        prefix,
        helperText,
        required,
        label,
        onChange,
        root: react_utilities_1.getNativeElementProps('div', {
            ref,
            ...props,
        }),
        components: {
            root: 'div',
            textFieldBorder: 'fieldset',
            textFieldLegend: 'legend',
            textFieldHelperText: 'p',
            textFieldWrapper: 'div',
            textFieldLabel: 'label',
            textFieldSuffix: 'div',
            textFieldPrefix: 'div',
            inputWrapper: 'div',
            input: 'input',
        },
        textFieldBorder: react_utilities_1.resolveShorthand(textFieldBorder, { required: true }),
        textFieldLegend: react_utilities_1.resolveShorthand(textFieldLegend, { required: true }),
        textFieldWrapper: react_utilities_1.resolveShorthand(textFieldWrapper, { required: true }),
        textFieldSuffix: react_utilities_1.resolveShorthand(textFieldSuffix, { required: true }),
        textFieldPrefix: react_utilities_1.resolveShorthand(textFieldPrefix, { required: true }),
        textFieldLabel: react_utilities_1.resolveShorthand(textFieldLabel, { required: true }),
        textFieldHelperText: react_utilities_1.resolveShorthand(textFieldHelperText, { required: true }),
        inputWrapper: react_utilities_1.resolveShorthand(inputWrapper, { required: true }),
        input: react_utilities_1.resolveShorthand(input, {
            required: true,
            defaultProps: {
                type: number ? 'number' : 'text',
            },
        }),
    };
    useTextFieldState_1.useTextFieldState(state);
    return state;
};
exports.useTextField = useTextField;
