// @pongoai/react-rating/lib/components/Rating/useRating.js
import { getNativeElementProps, resolveShorthand, useId } from '@fluentui/react-utilities';
import { useRatingState } from './useRatingState';
export const ratingShorthandProps = ['root', 'starWrapper', 'input'];
export const useRating = (props, ref) => {
    const { 
    // Props
    value, defaultValue = 0, max = 5, disabled = false, step = 1, readOnly = false, size = 'medium', onChange, 
    // Slots
    starWrapper, input, } = props;
    const state = {
        value,
        defaultValue,
        max,
        disabled,
        step,
        readOnly,
        onChange,
        size,
        components: {
            root: 'span',
            starWrapper: 'span',
            input: 'input',
        },
        root: getNativeElementProps('span', {
            ref,
            ...props,
            id: useId('rating-', props.id),
        }),
        starWrapper: resolveShorthand(starWrapper, { required: true }),
        input: resolveShorthand(input, {
            required: true,
            defaultProps: {
                type: 'range',
            },
        }),
    };
    useRatingState(state);
    return state;
};
