// @pongoai/react-theme/lib/themes/web/webLightTokens.js
export const webLightTokens = {
    base: '#000000',
    neutral: '#FFFFFF',
    inherit: '#4d4d4d',
    inheritHover: '#3e3e3e',
    inheritPressed: '#363636',
    inheritDisabled: '#a6a6a6',
    inheritForegroundHover: '#f5f5f5',
    inheritForegroundPressed: '#ebebeb',
    inheritBackground: '#dbdbdb',
    inheritBackgroundDisabled: '#f1f1f1',
    brand: '#2060CF',
    brandHover: '#1a4da6',
    brandPressed: '#164391',
    brandDisabled: '#8fafe7',
    brandForegroundHover: '#f6f9fd',
    brandForegroundPressed: '#eef3fb',
    brandBackground: '#d9e4f7',
    brandBackgroundHover: '#d0dbe',
    brandBackgroundPressed: '#c8d2e3',
    secondary: '#586bd9',
    secondaryHover: '#4656ae',
    secondaryPressed: '#3e4b98',
    secondaryDisabled: '#abb5ec',
    secondaryForegroundHover: '#f8f9fe',
    secondaryForegroundPressed: '#f2f4fc',
    success: '#278536',
    successHover: '#1f6a2b',
    successPressed: '#1b5d26',
    successDisabled: '#93c29a',
    successForegroundHover: '#f7faf7',
    successForegroundPressed: '#eef6ef',
    error: '#de3309',
    errorHover: '#b22907',
    errorPressed: '#9c2406',
    errorDisabled: '#ee9984',
    errorForegroundHover: '#fef7f5',
    errorForegroundPressed: '#fdefec',
    social: '#1ba1fb',
    socialHover: '#1681c9',
    socialPressed: '#1371b0',
    socialDisabled: '#8dd0fd',
    socialForegroundHover: '#f6fbff',
    socialForegroundPressed: '#edf8ff',
    warning: '#f18e21',
    warningHover: '#c1721a',
    warningPressed: '#a96417',
    warningDisabled: '#f8c690',
    warningForegroundHover: '#fffbf6',
    warningForegroundPressed: '#fef6ee',
    info: '#1da8b5',
    infoHover: '#1ca1ae',
    infoPressed: '#14767f',
    infoDisabled: '#8ed3da',
    infoForegroundHover: '#f6fcfc',
    infoForegroundPressed: '#edf8f9',
};
