// @pongoai/react-switch/lib-commonjs/Switch.stories.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.BasicSwitchExample = void 0;
const React = __importStar(require("react"));
const index_1 = require("./components/Switch/index");
const react_make_styles_1 = require("@fluentui/react-make-styles");
const useStyles = react_make_styles_1.makeStyles({
    root: {
        display: 'inline-flex',
        flexDirection: 'column',
        gap: '10px',
    },
    row: {
        display: 'inline-flex',
        flexDirection: 'row',
        gap: '10px',
    },
});
const BasicSwitchExample = () => {
    const styles = useStyles();
    return (React.createElement("div", { className: styles.root },
        React.createElement("div", { className: styles.row },
            React.createElement(index_1.Switch, null),
            React.createElement(index_1.Switch, { defaultChecked: true })),
        React.createElement("div", { className: styles.row },
            React.createElement(index_1.Switch, { disabled: true }),
            React.createElement(index_1.Switch, { defaultChecked: true, disabled: true }))));
};
exports.BasicSwitchExample = BasicSwitchExample;
exports.default = {
    title: 'Components/Switch',
    component: index_1.Switch,
};
