// @pongoai/react-link/lib-commonjs/Link.stories.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.BasicLinkExample = void 0;
const React = __importStar(require("react"));
const index_1 = require("./index");
const react_text_1 = require("@pongoai/react-text");
const react_make_styles_1 = require("@fluentui/react-make-styles");
const useStyles = react_make_styles_1.makeStyles({
    root: {
        display: 'inline-flex',
        flexDirection: 'column',
        gap: '5px',
    },
    row: {
        display: 'inline-flex',
        flexDirection: 'column',
        gap: '10px',
    },
});
const BasicLinkExample = () => {
    const styles = useStyles();
    return (React.createElement("div", { className: styles.root },
        React.createElement(react_text_1.Header2, null, "Basic Example"),
        React.createElement("div", { className: styles.row },
            React.createElement(index_1.Link, null, "Basic"),
            React.createElement(index_1.Link, { disabled: true }, "Basic"),
            React.createElement(index_1.Link, { disabled: true }, "Basic"))));
};
exports.BasicLinkExample = BasicLinkExample;
exports.default = {
    title: 'Components/Link',
    component: index_1.Link,
};
