// @fluentui/react-theme/lib-commonjs/global/utils.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.createGlobalTheme = void 0;

var tslib_1 = /*#__PURE__*/require("tslib");

var borderRadius_1 = /*#__PURE__*/require("./borderRadius");

var colors_1 = /*#__PURE__*/require("./colors");

var fonts_1 = /*#__PURE__*/require("./fonts");

var strokeWidths_1 = /*#__PURE__*/require("./strokeWidths");

var createGlobalTheme = function (brand, globalSharedColors) {
  return {
    color: {
      white: colors_1.white,
      black: colors_1.black,
      hcHyperlink: colors_1.hcHyperlink,
      hcHighlight: colors_1.hcHighlight,
      hcDisabled: colors_1.hcDisabled,
      hcCanvas: colors_1.hcCanvas,
      hcCanvasText: colors_1.hcCanvasText,
      hcHighlightText: colors_1.hcHighlightText,
      hcButtonText: colors_1.hcButtonText,
      hcButtonFace: colors_1.hcButtonFace
    },
    palette: tslib_1.__assign(tslib_1.__assign({}, globalSharedColors), {
      brand: brand,
      grey: colors_1.grey,
      whiteAlpha: colors_1.whiteAlpha,
      blackAlpha: colors_1.blackAlpha
    }),
    type: {
      alignment: fonts_1.textAlignments,
      fontFamilies: fonts_1.fontFamilies,
      fontSizes: fonts_1.fontSizes,
      fontWeights: fonts_1.fontWeights,
      lineHeights: fonts_1.lineHeights
    },
    borderRadius: borderRadius_1.borderRadius,
    strokeWidth: strokeWidths_1.strokeWidths
  };
};

exports.createGlobalTheme = createGlobalTheme;
