// @fluentui/keyboard-keys/lib/keyCodes.js
export var Cancel = 3;
export var Help = 6;
export var Backspace = 8;
export var Tab = 9;
export var Clear = 12;
export var Enter = 13;
export var Shift = 16;
export var Control = 17;
export var Alt = 18;
export var Pause = 19;
export var CapsLock = 20;
export var Escape = 27;
export var Convert = 28;
export var NonConvert = 29;
export var Accept = 30;
export var ModeChange = 31;
export var Space = 32;
export var PageUp = 33;
export var PageDown = 34;
export var End = 35;
export var Home = 36;
export var ArrowLeft = 37;
export var ArrowUp = 38;
export var ArrowRight = 39;
export var ArrowDown = 40;
export var Select = 41;
export var Print = 42;
export var Execute = 43;
export var PrintScreen = 44;
export var Insert = 45;
export var Delete = 46;
export var Digit0 = 48;
export var RightParenthesis = 48;
export var Digit1 = 49;
export var ExclamationPoint = 49;
export var Digit2 = 50;
export var AtSign = 50;
export var Digit3 = 51;
export var PoundSign = 51;
export var Digit4 = 52;
export var DollarSign = 52;
export var Digit5 = 53;
export var PercentSign = 53;
export var Digit6 = 54;
export var Caret = 54;
export var Digit7 = 55;
export var Ampersand = 55;
export var Digit8 = 56;
export var MultiplicationSign = 56;
export var Digit9 = 57;
export var LeftParenthesis = 57;
export var a = 65;
export var A = 65;
export var b = 66;
export var B = 66;
export var c = 67;
export var C = 67;
export var d = 68;
export var D = 68;
export var e = 69;
export var E = 69;
export var f = 70;
export var F = 70;
export var g = 71;
export var G = 71;
export var h = 72;
export var H = 72;
export var i = 73;
export var I = 73;
export var j = 74;
export var J = 74;
export var k = 75;
export var K = 75;
export var l = 76;
export var L = 76;
export var m = 77;
export var M = 77;
export var n = 78;
export var N = 78;
export var o = 79;
export var O = 79;
export var p = 80;
export var P = 80;
export var q = 81;
export var Q = 81;
export var r = 82;
export var R = 82;
export var s = 83;
export var S = 83;
export var t = 84;
export var T = 84;
export var u = 85;
export var U = 85;
export var v = 86;
export var V = 86;
export var w = 87;
export var W = 87;
export var x = 88;
export var X = 88;
export var y = 89;
export var Y = 89;
export var z = 90;
export var Z = 90;
export var OS = 91;
export var ContextMenu = 93;
export var F1 = 112;
export var F2 = 113;
export var F3 = 114;
export var F4 = 115;
export var F5 = 116;
export var F6 = 117;
export var F7 = 118;
export var F8 = 119;
export var F9 = 120;
export var F10 = 121;
export var F11 = 122;
export var F12 = 123;
export var F13 = 124;
export var F14 = 125;
export var F15 = 126;
export var F16 = 127;
export var F17 = 128;
export var F18 = 129;
export var F19 = 130;
export var F20 = 131;
export var F21 = 132;
export var F22 = 133;
export var F23 = 134;
export var F24 = 135;
export var NumLock = 144;
export var ScrollLock = 145;
export var VolumeMute = 181;
export var VolumeDown = 182;
export var VolumeUp = 183;
export var Semicolon = 186;
export var EqualsSign = 187;
export var PlusSign = 187;
export var Comma = 188;
export var LeftAngleBracket = 188;
export var MinusSign = 189;
export var Underscore = 189;
export var Decimal = 190;
export var RightAngleBracket = 190;
export var ForwardSlash = 191;
export var QuestionMark = 191;
export var GraveAccent = 192;
export var Tilde = 192;
export var LeftSquareBracket = 219;
export var LeftCurlyBrace = 219;
export var BackSlash = 220;
export var Pipe = 220;
export var RightSquareBracket = 221;
export var RightCurlyBrace = 221;
export var SingleQuote = 222;
export var DoubleQuote = 222;
export var Meta = 224;
export var AltGraph = 225;
export var Attn = 246;
export var CrSel = 247;
export var ExSel = 248;
export var EraseEof = 249;
export var Play = 250;
export var ZoomOut = 251;
