// @fluentui/keyboard-keys/lib/keys.js
export var Alt = 'Alt';
export var AltGraph = 'AltGraph';
export var CapsLock = 'CapsLock';
export var Control = 'Control';
export var Fn = 'Fn';
export var FnLock = 'FnLock';
export var Meta = 'Meta';
export var NumLock = 'NumLock';
export var ScrollLock = 'ScrollLock';
export var Shift = 'Shift';
export var Symbol = 'Symbol';
export var SymbolLock = 'SymbolLock';
export var Hyper = 'Hyper';
export var Super = 'Super';
export var Enter = 'Enter';
export var Space = ' ';
export var Tab = 'Tab';
export var ArrowDown = 'ArrowDown';
export var ArrowLeft = 'ArrowLeft';
export var ArrowRight = 'ArrowRight';
export var ArrowUp = 'ArrowUp';
export var End = 'End';
export var Home = 'Home';
export var PageDown = 'PageDown';
export var PageUp = 'PageUp';
export var Backspace = 'Backspace';
export var Clear = 'Clear';
export var Copy = 'Copy';
export var CrSel = 'CrSel';
export var Cut = 'Cut';
export var Delete = 'Delete';
export var EraseEof = 'EraseEof';
export var ExSel = 'ExSel';
export var Insert = 'Insert';
export var Paste = 'Paste';
export var Redo = 'Redo';
export var Undo = 'Undo';
export var Accept = 'Accept';
export var Again = 'Again';
export var Attn = 'Attn';
export var Cancel = 'Cancel';
export var ContextMenu = 'ContextMenu';
export var Escape = 'Escape';
export var Execute = 'Execute';
export var Find = 'Find';
export var Help = 'Help';
export var Pause = 'Pause';
export var Play = 'Play';
export var Props = 'Props';
export var Select = 'Select';
export var ZoomIn = 'ZoomIn';
export var ZoomOut = 'ZoomOut';
export var BrightnessDown = 'BrightnessDown';
export var BrightnessUp = 'BrightnessUp';
export var Eject = 'Eject';
export var LogOff = 'LogOff';
export var Power = 'Power';
export var PowerOff = 'PowerOff';
export var PrintScreen = 'PrintScreen';
export var Hibernate = 'Hibernate';
export var Standby = 'Standby';
export var WakeUp = 'WakeUp';
export var AllCandidates = 'AllCandidates';
export var Alphanumeric = 'Alphanumeric';
export var CodeInput = 'CodeInput';
export var Compose = 'Compose';
export var Convert = 'Convert';
export var Dead = 'Dead';
export var FinalMode = 'FinalMode';
export var GroupFirst = 'GroupFirst';
export var GroupLast = 'GroupLast';
export var GroupNext = 'GroupNext';
export var GroupPrevious = 'GroupPrevious';
export var ModeChange = 'ModeChange';
export var NextCandidate = 'NextCandidate';
export var NonConvert = 'NonConvert';
export var PreviousCandidate = 'PreviousCandidate';
export var Process = 'Process';
export var SingleCandidate = 'SingleCandidate';
export var HangulMode = 'HangulMode';
export var HanjaMode = 'HanjaMode';
export var JunjaMode = 'JunjaMode';
export var Eisu = 'Eisu';
export var Hankaku = 'Hankaku';
export var Hiragana = 'Hiragana';
export var HiraganaKatakana = 'HiraganaKatakana';
export var KanaMode = 'KanaMode';
export var KanjiMode = 'KanjiMode';
export var Katakana = 'Katakana';
export var Romaji = 'Romaji';
export var Zenkaku = 'Zenkaku';
export var ZenkakuHankaku = 'ZenkakuHankaku';
export var F1 = 'F1';
export var F2 = 'F2';
export var F3 = 'F3';
export var F4 = 'F4';
export var F5 = 'F5';
export var F6 = 'F6';
export var F7 = 'F7';
export var F8 = 'F8';
export var F9 = 'F9';
export var F10 = 'F10';
export var F11 = 'F11';
export var F12 = 'F12';
export var Soft1 = 'Soft1';
export var Soft2 = 'Soft2';
export var Soft3 = 'Soft3';
export var Soft4 = 'Soft4';
export var ChannelDown = 'ChannelDown';
export var ChannelUp = 'ChannelUp';
export var Close = 'Close';
export var MailForward = 'MailForward';
export var MailReply = 'MailReply';
export var MailSend = 'MailSend';
export var MediaClose = 'MediaClose';
export var MediaFastForward = 'MediaFastForward';
export var MediaPause = 'MediaPause';
export var MediaPlay = 'MediaPlay';
export var MediaPlayPause = 'MediaPlayPause';
export var MediaRecord = 'MediaRecord';
export var MediaRewind = 'MediaRewind';
export var MediaStop = 'MediaStop';
export var MediaTrackNext = 'MediaTrackNext';
export var MediaTrackPrevious = 'MediaTrackPrevious';
export var New = 'New';
export var Open = 'Open';
export var Print = 'Print';
export var Save = 'Save';
export var SpellCheck = 'SpellCheck';
export var Key11 = 'Key11';
export var Key12 = 'Key12';
export var AudioBalanceLeft = 'AudioBalanceLeft';
export var AudioBalanceRight = 'AudioBalanceRight';
export var AudioBassBoostDown = 'AudioBassBoostDown';
export var AudioBassBoostToggle = 'AudioBassBoostToggle';
export var AudioBassBoostUp = 'AudioBassBoostUp';
export var AudioFaderFront = 'AudioFaderFront';
export var AudioFaderRear = 'AudioFaderRear';
export var AudioSurroundModeNext = 'AudioSurroundModeNext';
export var AudioTrebleDown = 'AudioTrebleDown';
export var AudioTrebleUp = 'AudioTrebleUp';
export var AudioVolumeDown = 'AudioVolumeDown';
export var AudioVolumeUp = 'AudioVolumeUp';
export var AudioVolumeMute = 'AudioVolumeMute';
export var MicrophoneToggle = 'MicrophoneToggle';
export var MicrophoneVolumeDown = 'MicrophoneVolumeDown';
export var MicrophoneVolumeUp = 'MicrophoneVolumeUp';
export var MicrophoneVolumeMute = 'MicrophoneVolumeMute';
export var SpeechCorrectionList = 'SpeechCorrectionList';
export var SpeechInputToggle = 'SpeechInputToggle';
export var LaunchApplication1 = 'LaunchApplication1';
export var LaunchApplication2 = 'LaunchApplication2';
export var LaunchCalendar = 'LaunchCalendar';
export var LaunchContacts = 'LaunchContacts';
export var LaunchMail = 'LaunchMail';
export var LaunchMediaPlayer = 'LaunchMediaPlayer';
export var LaunchMusicPlayer = 'LaunchMusicPlayer';
export var LaunchPhone = 'LaunchPhone';
export var LaunchScreenSaver = 'LaunchScreenSaver';
export var LaunchSpreadsheet = 'LaunchSpreadsheet';
export var LaunchWebBrowser = 'LaunchWebBrowser';
export var LaunchWebCam = 'LaunchWebCam';
export var LaunchWordProcessor = 'LaunchWordProcessor';
export var BrowserBack = 'BrowserBack';
export var BrowserFavorites = 'BrowserFavorites';
export var BrowserForward = 'BrowserForward';
export var BrowserHome = 'BrowserHome';
export var BrowserRefresh = 'BrowserRefresh';
export var BrowserSearch = 'BrowserSearch';
export var BrowserStop = 'BrowserStop';
export var AppSwitch = 'AppSwitch';
export var Call = 'Call';
export var Camera = 'Camera';
export var CameraFocus = 'CameraFocus';
export var EndCall = 'EndCall';
export var GoBack = 'GoBack';
export var GoHome = 'GoHome';
export var HeadsetHook = 'HeadsetHook';
export var LastNumberRedial = 'LastNumberRedial';
export var Notification = 'Notification';
export var MannerMode = 'MannerMode';
export var VoiceDial = 'VoiceDial';
export var TV = 'TV';
export var TV3DMode = 'TV3DMode';
export var TVAntennaCable = 'TVAntennaCable';
export var TVAudioDescription = 'TVAudioDescription';
export var TVAudioDescriptionMixDown = 'TVAudioDescriptionMixDown';
export var TVAudioDescriptionMixUp = 'TVAudioDescriptionMixUp';
export var TVContentsMenu = 'TVContentsMenu';
export var TVDataService = 'TVDataService';
export var TVInput = 'TVInput';
export var TVInputComponent1 = 'TVInputComponent1';
export var TVInputComponent2 = 'TVInputComponent2';
export var TVInputComposite1 = 'TVInputComposite1';
export var TVInputComposite2 = 'TVInputComposite2';
export var TVInputHDMI1 = 'TVInputHDMI1';
export var TVInputHDMI2 = 'TVInputHDMI2';
export var TVInputHDMI3 = 'TVInputHDMI3';
export var TVInputHDMI4 = 'TVInputHDMI4';
export var TVInputVGA1 = 'TVInputVGA1';
export var TVMediaContext = 'TVMediaContext';
export var TVNetwork = 'TVNetwork';
export var TVNumberEntry = 'TVNumberEntry';
export var TVPower = 'TVPower';
export var TVRadioService = 'TVRadioService';
export var TVSatellite = 'TVSatellite';
export var TVSatelliteBS = 'TVSatelliteBS';
export var TVSatelliteCS = 'TVSatelliteCS';
export var TVSatelliteToggle = 'TVSatelliteToggle';
export var TVTerrestrialAnalog = 'TVTerrestrialAnalog';
export var TVTerrestrialDigital = 'TVTerrestrialDigital';
export var TVTimer = 'TVTimer';
export var AVRInput = 'AVRInput';
export var AVRPower = 'AVRPower';
export var ColorF0Red = 'ColorF0Red';
export var ColorF1Green = 'ColorF1Green';
export var ColorF2Yellow = 'ColorF2Yellow';
export var ColorF3Blue = 'ColorF3Blue';
export var ColorF4Grey = 'ColorF4Grey';
export var ColorF5Brown = 'ColorF5Brown';
export var ClosedCaptionToggle = 'ClosedCaptionToggle';
export var Dimmer = 'Dimmer';
export var DisplaySwap = 'DisplaySwap';
export var DVR = 'DVR';
export var Exit = 'Exit';
export var FavoriteClear0 = 'FavoriteClear0';
export var FavoriteClear1 = 'FavoriteClear1';
export var FavoriteClear2 = 'FavoriteClear2';
export var FavoriteClear3 = 'FavoriteClear3';
export var FavoriteRecall0 = 'FavoriteRecall0';
export var FavoriteRecall1 = 'FavoriteRecall1';
export var FavoriteRecall2 = 'FavoriteRecall2';
export var FavoriteRecall3 = 'FavoriteRecall3';
export var FavoriteStore0 = 'FavoriteStore0';
export var FavoriteStore1 = 'FavoriteStore1';
export var FavoriteStore2 = 'FavoriteStore2';
export var FavoriteStore3 = 'FavoriteStore3';
export var Guide = 'Guide';
export var GuideNextDay = 'GuideNextDay';
export var GuidePreviousDay = 'GuidePreviousDay';
export var Info = 'Info';
export var InstantReplay = 'InstantReplay';
export var Link = 'Link';
export var ListProgram = 'ListProgram';
export var LiveContent = 'LiveContent';
export var Lock = 'Lock';
export var MediaApps = 'MediaApps';
export var MediaAudioTrack = 'MediaAudioTrack';
export var MediaLast = 'MediaLast';
export var MediaSkipBackward = 'MediaSkipBackward';
export var MediaSkipForward = 'MediaSkipForward';
export var MediaStepBackward = 'MediaStepBackward';
export var MediaStepForward = 'MediaStepForward';
export var MediaTopMenu = 'MediaTopMenu';
export var NavigateIn = 'NavigateIn';
export var NavigateNext = 'NavigateNext';
export var NavigateOut = 'NavigateOut';
export var NavigatePrevious = 'NavigatePrevious';
export var NextFavoriteChannel = 'NextFavoriteChannel';
export var NextUserProfile = 'NextUserProfile';
export var OnDemand = 'OnDemand';
export var Pairing = 'Pairing';
export var PinPDown = 'PinPDown';
export var PinPMove = 'PinPMove';
export var PinPToggle = 'PinPToggle';
export var PinPUp = 'PinPUp';
export var PlaySpeedDown = 'PlaySpeedDown';
export var PlaySpeedReset = 'PlaySpeedReset';
export var PlaySpeedUp = 'PlaySpeedUp';
export var RandomToggle = 'RandomToggle';
export var RcLowBattery = 'RcLowBattery';
export var RecordSpeedNext = 'RecordSpeedNext';
export var RfBypass = 'RfBypass';
export var ScanChannelsToggle = 'ScanChannelsToggle';
export var ScreenModeNext = 'ScreenModeNext';
export var Settings = 'Settings';
export var SplitScreenToggle = 'SplitScreenToggle';
export var STBInput = 'STBInput';
export var STBPower = 'STBPower';
export var Subtitle = 'Subtitle';
export var Teletext = 'Teletext';
export var VideoModeNext = 'VideoModeNext';
export var Wink = 'Wink';
export var ZoomToggle = 'ZoomToggle';
export var MediaNextTrack = 'MediaNextTrack';
export var MediaPreviousTrack = 'MediaPreviousTrack';
export var Unidentified = 'Unidentified';
