// @fluentui/react-make-styles/lib-commonjs/makeStyles.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.makeStyles = void 0;

var make_styles_1 = /*#__PURE__*/require("@fluentui/make-styles");

var react_shared_contexts_1 = /*#__PURE__*/require("@fluentui/react-shared-contexts");

var React = /*#__PURE__*/require("react");

var RendererContext_1 = /*#__PURE__*/require("./RendererContext");

function isInsideComponent() {
  try {
    // eslint-disable-next-line react-hooks/rules-of-hooks
    React.useContext({});
    return true;
  } catch (e) {
    return false;
  }
}

function makeStyles(stylesBySlots) {
  var getStyles = make_styles_1.makeStyles(stylesBySlots);

  if (process.env.NODE_ENV !== 'production') {
    if (isInsideComponent()) {
      throw new Error(["makeStyles(): this function cannot be called in component's scope.", 'All makeStyles() calls should be top level i.e. in a root scope of a file.'].join(' '));
    }
  }

  return function useClasses() {
    var dir = react_shared_contexts_1.useFluent().dir;
    var renderer = RendererContext_1.useRenderer();
    var options = {
      dir: dir,
      renderer: renderer
    };
    return getStyles(options);
  };
}

exports.makeStyles = makeStyles;
