// @pongoai/react-rating/lib-commonjs/components/Rating/useRating.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.useRating = exports.ratingShorthandProps = void 0;
const react_utilities_1 = require("@fluentui/react-utilities");
const useRatingState_1 = require("./useRatingState");
exports.ratingShorthandProps = ['root', 'starWrapper', 'input'];
const useRating = (props, ref) => {
    const { 
    // Props
    value, defaultValue = 0, max = 5, disabled = false, step = 1, readOnly = false, size = 'medium', onChange, 
    // Slots
    starWrapper, input, } = props;
    const state = {
        value,
        defaultValue,
        max,
        disabled,
        step,
        readOnly,
        onChange,
        size,
        components: {
            root: 'span',
            starWrapper: 'span',
            input: 'input',
        },
        root: react_utilities_1.getNativeElementProps('span', {
            ref,
            ...props,
            id: react_utilities_1.useId('rating-', props.id),
        }),
        starWrapper: react_utilities_1.resolveShorthand(starWrapper, { required: true }),
        input: react_utilities_1.resolveShorthand(input, {
            required: true,
            defaultProps: {
                type: 'range',
            },
        }),
    };
    useRatingState_1.useRatingState(state);
    return state;
};
exports.useRating = useRating;
