// @pongoai/react-textfield/lib-commonjs/components/Select/useSelect.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.useSelect = exports.selectShorthandProps = void 0;
const react_utilities_1 = require("@fluentui/react-utilities");
const useSelectState_1 = require("./useSelectState");
exports.selectShorthandProps = [
    'root',
    'selectButton',
    'textFieldLabel',
    'list',
    'textFieldLegend',
    'textFieldHelperText',
    'textFieldWrapper',
    'textFieldSuffix',
    'textFieldPrefix',
    'input',
    'inputWrapper',
    'textFieldBorder',
];
const useSelect = (props, ref) => {
    const { defaultValue = '', value, disabled = false, error = false, appearance = 'standard', placeholder, options, required = false, label, helperText, inputWrapper, input, textFieldHelperText, textFieldLabel, textFieldLegend, textFieldBorder, textFieldWrapper, textFieldSuffix, textFieldPrefix, suffix, prefix, selectButton, list, onChange, } = props;
    const state = {
        defaultValue,
        value,
        disabled,
        options,
        error,
        appearance,
        placeholder,
        helperText,
        required,
        suffix,
        prefix,
        label,
        onChange,
        root: react_utilities_1.getNativeElementProps('div', {
            ref,
            ...props,
        }),
        components: {
            root: 'div',
            textFieldBorder: 'fieldset',
            textFieldLegend: 'legend',
            textFieldHelperText: 'p',
            textFieldWrapper: 'div',
            textFieldLabel: 'label',
            selectButton: 'div',
            textFieldSuffix: 'div',
            textFieldPrefix: 'div',
            inputWrapper: 'div',
            input: 'input',
            list: 'ul',
        },
        textFieldBorder: react_utilities_1.resolveShorthand(textFieldBorder, { required: true }),
        textFieldLegend: react_utilities_1.resolveShorthand(textFieldLegend, { required: true }),
        textFieldWrapper: react_utilities_1.resolveShorthand(textFieldWrapper, { required: true }),
        textFieldLabel: react_utilities_1.resolveShorthand(textFieldLabel, { required: true }),
        textFieldHelperText: react_utilities_1.resolveShorthand(textFieldHelperText, { required: true }),
        textFieldSuffix: react_utilities_1.resolveShorthand(textFieldSuffix, { required: true }),
        textFieldPrefix: react_utilities_1.resolveShorthand(textFieldPrefix, { required: true }),
        selectButton: react_utilities_1.resolveShorthand(selectButton, { required: true }),
        list: react_utilities_1.resolveShorthand(list, { required: true }),
        inputWrapper: react_utilities_1.resolveShorthand(inputWrapper, { required: true }),
        input: react_utilities_1.resolveShorthand(input, {
            required: true,
            defaultProps: {
                type: 'text',
            },
        }),
    };
    useSelectState_1.useSelectState(state);
    return state;
};
exports.useSelect = useSelect;
