// @pongoai/react-textfield/lib-commonjs/components/Multiline/useMultiline.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.useMultiline = exports.multilineShorthandProps = void 0;
const react_utilities_1 = require("@fluentui/react-utilities");
const useMultilineState_1 = require("./useMultilineState");
exports.multilineShorthandProps = [
    'root',
    'textarea',
    'textFieldLabel',
    'textFieldLegend',
    'textFieldHelperText',
    'textFieldBorder',
];
const useMultiline = (props, ref) => {
    const { defaultValue = '', value, autocomplete, disabled = false, error = false, appearance = 'standard', textarea, placeholder, required = false, autoAdjust = true, label, helperText, textFieldHelperText, textFieldLabel, textFieldLegend, textFieldBorder, resizable = false, onChange, } = props;
    const state = {
        defaultValue,
        value,
        autocomplete,
        disabled,
        error,
        appearance,
        placeholder,
        resizable,
        helperText,
        autoAdjust,
        required,
        label,
        onChange,
        root: react_utilities_1.getNativeElementProps('div', {
            ref,
            ...props,
        }),
        components: {
            root: 'div',
            textFieldBorder: 'fieldset',
            textFieldLegend: 'legend',
            textFieldHelperText: 'p',
            textFieldLabel: 'label',
            textarea: 'textarea',
        },
        textFieldBorder: react_utilities_1.resolveShorthand(textFieldBorder, { required: true }),
        textFieldLegend: react_utilities_1.resolveShorthand(textFieldLegend, { required: true }),
        textFieldLabel: react_utilities_1.resolveShorthand(textFieldLabel, { required: true }),
        textFieldHelperText: react_utilities_1.resolveShorthand(textFieldHelperText, { required: true }),
        textarea: react_utilities_1.resolveShorthand(textarea, {
            required: true,
        }),
    };
    useMultilineState_1.useMultilineState(state);
    return state;
};
exports.useMultiline = useMultiline;
