// @pongoai/react-textfield/lib/Multiline.stories.js
import * as React from 'react';
import { Multiline, TextField } from './index';
import { Header2 } from '@pongoai/react-text';
import { makeStyles } from '@fluentui/react-make-styles';
const useStyles = makeStyles({
    root: {
        display: 'inline-flex',
        flexDirection: 'column',
        gap: '5px',
    },
    row: {
        display: 'inline-flex',
        flexDirection: 'row',
        gap: '10px',
    },
});
export const BasicMultilineExample = () => {
    const styles = useStyles();
    return (React.createElement("div", { className: styles.root },
        React.createElement(Header2, null, "Standard Multiline Text Field"),
        React.createElement("div", { className: styles.row },
            React.createElement(Multiline, { label: "Standard", placeholder: "Enter your text" }),
            React.createElement(Multiline, { label: "Standard", placeholder: "Enter your text", disabled: true }),
            React.createElement(Multiline, { label: "Standard", placeholder: "Enter your text", helperText: "Error", error: true })),
        React.createElement(Header2, null, "Outlined Multiline Text Field"),
        React.createElement("div", { className: styles.row },
            React.createElement(Multiline, { appearance: "outlined", placeholder: "Enter your text", label: "Outlined" }),
            React.createElement(TextField, { appearance: "outlined", label: "Outlined", placeholder: "Enter your text" }),
            React.createElement(Multiline, { appearance: "outlined", placeholder: "Enter your text", label: "Outlined", disabled: true }),
            React.createElement(Multiline, { appearance: "outlined", label: "Standard", placeholder: "Enter your text", helperText: "Error", error: true })),
        React.createElement(Header2, null, "Filled Multiline Text Field"),
        React.createElement("div", { className: styles.row },
            React.createElement(Multiline, { appearance: "filled", placeholder: "Enter your text", label: "Filled" }),
            React.createElement(Multiline, { appearance: "filled", placeholder: "Enter your text", label: "Filled", disabled: true }),
            React.createElement(Multiline, { appearance: "filled", label: "Standard", placeholder: "Enter your text", helperText: "Error", error: true }))));
};
export default {
    title: 'Components/Multiline',
    component: Multiline,
};
