// @pongoai/react-textfield/lib/Password.stories.js
import * as React from 'react';
import { Password } from './index';
import { Header2 } from '@pongoai/react-text';
import { makeStyles } from '@fluentui/react-make-styles';
const useStyles = makeStyles({
    root: {
        display: 'inline-flex',
        flexDirection: 'column',
        gap: '5px',
    },
    row: {
        display: 'inline-flex',
        flexDirection: 'row',
        gap: '10px',
    },
});
export const BasicPasswordExample = () => {
    const styles = useStyles();
    return (React.createElement("div", { className: styles.root },
        React.createElement(Header2, null, "Password Text Field"),
        React.createElement("div", { className: styles.row },
            React.createElement(Password, { appearance: "standard", placeholder: "Enter your password", label: "Password" }),
            React.createElement(Password, { appearance: "outlined", placeholder: "Enter your password", label: "Password" }),
            React.createElement(Password, { appearance: "filled", placeholder: "Enter your password", label: "Password" }))));
};
export default {
    title: 'Components/Password',
    component: Password,
};
