// @pongoai/react-textfield/lib/Search.stories.js
import * as React from 'react';
import { Search } from './index';
import { Header2 } from '@pongoai/react-text';
import { makeStyles } from '@fluentui/react-make-styles';
const useStyles = makeStyles({
    root: {
        display: 'inline-flex',
        flexDirection: 'column',
        gap: '5px',
    },
    row: {
        display: 'inline-flex',
        flexDirection: 'row',
        gap: '10px',
    },
});
export const BasicSearchExample = () => {
    const styles = useStyles();
    return (React.createElement("div", { className: styles.root },
        React.createElement(Header2, null, "Search Text Field"),
        React.createElement("div", { className: styles.row },
            React.createElement(Search, { appearance: "standard", placeholder: "Type to find...", label: "Search surveys" }),
            React.createElement(Search, { appearance: "outlined", placeholder: "Type to find...", label: "Search surveys" }),
            React.createElement(Search, { appearance: "filled", placeholder: "Type to find...", label: "Search surveys" }))));
};
export default {
    title: 'Components/Search',
    component: Search,
};
