// @fluentui/react-theme/lib/alias/light.js
import { __assign } from "tslib";
import { black, grey, white, sharedColors } from '../global/colors';
export var generateColorTokens = function (brand) {
  return {
    colorNeutralForeground1: grey[14],
    colorNeutralForeground2: grey[26],
    colorNeutralForeground2Hover: grey[14],
    colorNeutralForeground2Pressed: grey[14],
    colorNeutralForeground2Selected: grey[14],
    colorNeutralForeground2BrandHover: brand.primary,
    colorNeutralForeground2BrandPressed: brand.shade10,
    colorNeutralForeground2BrandSelected: brand.primary,
    colorNeutralForeground3: grey[38],
    colorNeutralForeground3Hover: grey[26],
    colorNeutralForeground3Pressed: grey[26],
    colorNeutralForeground3Selected: grey[26],
    colorNeutralForeground3BrandHover: brand.primary,
    colorNeutralForeground3BrandPressed: brand.shade10,
    colorNeutralForeground3BrandSelected: brand.primary,
    colorNeutralForeground4: grey[44],
    colorNeutralForegroundDisabled: grey[74],
    colorBrandForegroundLink: brand.shade10,
    colorBrandForegroundLinkHover: brand.shade20,
    colorBrandForegroundLinkPressed: brand.shade40,
    colorBrandForegroundLinkSelected: brand.shade10,
    colorCompoundBrandForeground1: brand.primary,
    colorCompoundBrandForeground1Hover: brand.shade10,
    colorCompoundBrandForeground1Pressed: brand.shade20,
    colorBrandForeground1: brand.primary,
    colorBrandForeground2: brand.shade10,
    colorNeutralForegroundInverted: white,
    colorNeutralForegroundOnBrand: white,
    colorNeutralForegroundInvertedLink: white,
    colorNeutralForegroundInvertedLinkHover: white,
    colorNeutralForegroundInvertedLinkPressed: white,
    colorNeutralForegroundInvertedLinkSelected: white,
    colorNeutralBackground1: white,
    colorNeutralBackground1Hover: grey[96],
    colorNeutralBackground1Pressed: grey[88],
    colorNeutralBackground1Selected: grey[92],
    colorNeutralBackground2: grey[98],
    colorNeutralBackground2Hover: grey[94],
    colorNeutralBackground2Pressed: grey[86],
    colorNeutralBackground2Selected: grey[90],
    colorNeutralBackground3: grey[96],
    colorNeutralBackground3Hover: grey[92],
    colorNeutralBackground3Pressed: grey[84],
    colorNeutralBackground3Selected: grey[88],
    colorNeutralBackground4: grey[94],
    colorNeutralBackground4Hover: grey[98],
    colorNeutralBackground4Pressed: grey[96],
    colorNeutralBackground4Selected: white,
    colorNeutralBackground5: grey[92],
    colorNeutralBackground5Hover: grey[96],
    colorNeutralBackground5Pressed: grey[94],
    colorNeutralBackground5Selected: grey[98],
    colorNeutralBackground6: grey[90],
    colorNeutralBackgroundInverted: grey[38],
    colorSubtleBackground: 'transparent',
    colorSubtleBackgroundHover: grey[96],
    colorSubtleBackgroundPressed: grey[88],
    colorSubtleBackgroundSelected: grey[92],
    colorTransparentBackground: 'transparent',
    colorTransparentBackgroundHover: 'transparent',
    colorTransparentBackgroundPressed: 'transparent',
    colorTransparentBackgroundSelected: 'transparent',
    colorNeutralBackgroundDisabled: grey[94],
    colorNeutralStencil1: grey[90],
    colorNeutralStencil2: grey[98],
    colorBrandBackground: brand.primary,
    colorBrandBackgroundHover: brand.shade10,
    colorBrandBackgroundPressed: brand.shade40,
    colorBrandBackgroundSelected: brand.shade20,
    colorCompoundBrandBackground: brand.primary,
    colorCompoundBrandBackgroundHover: brand.shade10,
    colorCompoundBrandBackgroundPressed: brand.shade20,
    colorBrandBackgroundStatic: brand.primary,
    colorBrandBackground2: brand.tint60,
    colorNeutralStrokeAccessible: grey[38],
    colorNeutralStrokeAccessibleHover: grey[34],
    colorNeutralStrokeAccessiblePressed: grey[30],
    colorNeutralStrokeAccessibleSelected: brand.primary,
    colorNeutralStroke1: grey[82],
    colorNeutralStroke1Hover: grey[78],
    colorNeutralStroke1Pressed: grey[70],
    colorNeutralStroke1Selected: grey[74],
    colorNeutralStroke2: grey[88],
    colorNeutralStroke3: grey[94],
    colorBrandStroke1: brand.primary,
    colorBrandStroke2: brand.tint40,
    colorCompoundBrandStroke: brand.primary,
    colorCompoundBrandStrokeHover: brand.shade10,
    colorCompoundBrandStrokePressed: brand.shade20,
    colorNeutralStrokeDisabled: grey[88],
    colorTransparentStroke: 'transparent',
    colorTransparentStrokeInteractive: 'transparent',
    colorTransparentStrokeDisabled: 'transparent',
    colorStrokeFocus1: white,
    colorStrokeFocus2: black,
    colorNeutralShadowAmbient: 'rgba(0,0,0,0.12)',
    colorNeutralShadowKey: 'rgba(0,0,0,0.14)',
    colorNeutralShadowAmbientLighter: 'rgba(0,0,0,0.06)',
    colorNeutralShadowKeyLighter: 'rgba(0,0,0,0.07)',
    colorNeutralShadowAmbientDarker: 'rgba(0,0,0,0.20)',
    colorNeutralShadowKeyDarker: 'rgba(0,0,0,0.24)',
    colorBrandShadowAmbient: 'rgba(0,0,0,0.30)',
    colorBrandShadowKey: 'rgba(0,0,0,0.25)'
  };
};
export var colorPaletteTokens = /*#__PURE__*/Object.keys(sharedColors).reduce(function (acc, sharedColor) {
  var _a;

  var color = sharedColor.slice(0, 1).toUpperCase() + sharedColor.slice(1);
  var sharedColorTokens = (_a = {}, _a["colorPalette" + color + "Background1"] = sharedColors[sharedColor].tint60, _a["colorPalette" + color + "Background2"] = sharedColors[sharedColor].tint40, _a["colorPalette" + color + "Background3"] = sharedColors[sharedColor].primary, _a["colorPalette" + color + "Foreground1"] = sharedColors[sharedColor].shade10, _a["colorPalette" + color + "Foreground2"] = sharedColors[sharedColor].shade30, _a["colorPalette" + color + "Foreground3"] = sharedColors[sharedColor].primary, _a["colorPalette" + color + "BorderActive"] = sharedColors[sharedColor].primary, _a["colorPalette" + color + "Border2"] = sharedColors[sharedColor].primary, _a);
  return __assign(__assign({}, acc), sharedColorTokens);
}, {});
