// @pongoai/react-rating/lib-commonjs/Rating.stories.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.BasicRatingExample = void 0;
const React = __importStar(require("react"));
const index_1 = require("./index");
const react_text_1 = require("@pongoai/react-text");
const react_make_styles_1 = require("@fluentui/react-make-styles");
const useStyles = react_make_styles_1.makeStyles({
    root: {
        display: 'inline-flex',
        flexDirection: 'column',
        gap: '10px',
    },
    row: {
        display: 'inline-flex',
        flexDirection: 'column',
        gap: '10px',
    },
});
const BasicRatingExample = () => {
    const [ratingValue, setRatingValue] = React.useState(3);
    const styles = useStyles();
    const onRatingChange = (_ev, data) => {
        setRatingValue(data.value);
    };
    return (React.createElement("div", { className: styles.root },
        React.createElement(react_text_1.Header3, null, "Basic Example"),
        React.createElement("div", { className: styles.row },
            React.createElement(index_1.Rating, null)),
        React.createElement(react_text_1.Header3, null,
            "Controlled Example ",
            ratingValue),
        React.createElement("div", { className: styles.row },
            React.createElement(index_1.Rating, { value: ratingValue, onChange: onRatingChange })),
        React.createElement(react_text_1.Header3, null, "Read Only"),
        React.createElement("div", { className: styles.row },
            React.createElement(index_1.Rating, { defaultValue: 2, readOnly: true })),
        React.createElement(react_text_1.Header3, null, "Size"),
        React.createElement("div", { className: styles.row },
            React.createElement(index_1.Rating, { size: "small" }),
            React.createElement(index_1.Rating, { size: "medium" }),
            React.createElement(index_1.Rating, { size: "large" })),
        React.createElement(react_text_1.Header3, null, "Multiple Stars"),
        React.createElement("div", { className: styles.row },
            React.createElement(index_1.Rating, { defaultValue: 10, max: 10 })),
        React.createElement(react_text_1.Header3, null, "Disabled"),
        React.createElement("div", { className: styles.row },
            React.createElement(index_1.Rating, { defaultValue: 3, disabled: true }))));
};
exports.BasicRatingExample = BasicRatingExample;
exports.default = {
    title: 'Components/Rating',
    component: index_1.Rating,
};
